File: sem4/MPP/proiect/appClientServer/CompetitionModel/src/main/java/org/example/model/Participation.java
package org.example.model;

import java.io.Serializable;
import java.util.Objects;

public class Participation extends Entity<Integer> implements Serializable {
    private Child child;
    private Task task;

    public Participation(Child child, Task task) {
        this.child = child;
        this.task = task;
    }

    public Child getChild() {
        return child;
    }

    public void setChild(Child childId) {
        this.child = child;
    }

    public Task getTask() {
        return task;
    }

    public void setTask(Task task) {
        this.task = task;
    }

    @Override
    public String toString() {
        return "Participation{" +
                "child=" + child +
                ", task=" + task +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Participation that = (Participation) o;
        return child == that.child && task == that.task;
    }

    @Override
    public int hashCode() {
        return Objects.hash(child, task);
    }
}

File: sem4/MPP/proiectC#/AppClientServer/Model/Participation.cs
using System;

namespace model
{
    [Serializable]
    public class Participation: Entity<int>
    {
        public Child child { get; set; }
        public Task task { get; set; }

        public Participation(Child child, Task task)
        {
            this.child = child;
            this.task = task;
        }
        
        public Participation()
        { }

        public override bool Equals(object? obj)
        {
            if (obj is Participation)
            {
                return (obj as Participation).ID == ID;
            }
            return base.Equals(obj);
        }

        public override string ToString()
        {
            return ID + " " + child + " " + task;
        }
    
        public static bool operator ==(Participation p1, Participation p2)
        {
            return p1.Equals(p2);
        }
    
        public static bool operator !=(Participation p1, Participation p2)
        {
            return !p1.Equals(p2);
        }
    }
}
